app/models/concerns/sequenced.rb

```diff
@@ -5,8 +5,10 @@ module Sequenced
 
   included do
     scope :with_sequential_id, -> { where.not(sequential_id: nil) }
+    scope :with_org_sequential_id, -> { where.not(organization_sequential_id: nil) }
 
     before_save :ensure_sequential_id
+    before_save :ensure_organization_sequential_id, if: -> { self.class.to_s == 'Invoice' }
```

Comment:
Style/ClassEqualityComparison: Use `instance_of?(::Invoice)` instead of comparing classes. (https://rubystyle.guide#instance-of-vs-class-comparison)